### PR TITLE
setting up java dependencies

### PR DESCRIPTION
**Pull request by seunomonije**
Set up pom file to help with our java setup and to include any required dependences we might have. This is particularly useful for grabbing lyrics from the MusixMatch API, since they provide a link to a wrapper which will help alot. I'm sure that this will be of help to everyone else as well. 

Link to the wrapper: https://github.com/sachin-handiekar/jMusixMatch

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    <plugins>
+      <!-- Provides `mvn package appengine:run` for local testing
+           and `mvn package appengine:deploy` for deploying. -->
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>appengine-maven-plugin</artifactId>
+        <version>2.2.0</version>
+        <configuration>
+          <!-- TODO: set project ID. -->
+          <deploy.projectId></deploy.projectId>
```
Comment (laptou) [inline pom.xml:44]: Nick gave us a project ID at the end of yesterday, can you throw that in there please?
```suggestion
          <deploy.projectId>capstone-t99-2020</deploy.projectId>
```

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.google.sps</groupId>
+  <artifactId>portfolio</artifactId>
+  <version>1</version>
+  <packaging>war</packaging>
+
+  <properties>
+    <!-- Java 8. Potentially subject to change.-->
```
Comment (laptou) [inline pom.xml:12]: Nick didn't comment on whether we should use Java 8 or 11, I asked him in the chat and he didn't say anything so I don't think he cares. App Engine fully supports (and recommends) Java 11 in the configuration that we're going to be using it in, so could we upgrade it?

Comment (seunomonije) [reply pom.xml:12]: I'm down. Grace?

**Comment by laptou** [comment]
@seunomonije Accepting the patch created the commit under `ibiyemi@intulon.com`, which is not recognized by the CLA bot. I already signed the CLA as `ibiyemi@google.com`, so I'm not sure if I can fix this. Can you please delete that commit (`git reset` and force push) and then re-do it without including `ibiyemi@intulon.com` as a co-author (which is what happened automatically when you accepted the patch)?

**Comment by seunomonije** [comment]
@laptou ill just wait till it's merged since this branch probably won't last long anyways

File: pom.xml (hunk 2 above)
Comment (laptou) [reply pom.xml:12]: @gracetian6

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      <scope>provided</scope>
+    </dependency>
+
+    <!-- Gson dependency -->
+    <dependency>
+        <groupId>com.google.code.gson</groupId>
+        <artifactId>gson</artifactId>
+        <version>2.8.6</version>
+    </dependency>
+
```
Comment (laptou) [inline pom.xml:33]: Missing a closing `</dependencies>` tag.

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  <build>
+    <plugins>
+      <!-- Provides `mvn package appengine:run` for local testing
+           and `mvn package appengine:deploy` for deploying. -->
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>appengine-maven-plugin</artifactId>
+        <version>2.2.0</version>
+        <configuration>
+          <!-- TODO: set project ID. -->
```
Comment (laptou) [inline pom.xml:43]: Remove this

```suggestion
```

File: pom.xml (hunk 2 above)
Comment (gracetian6) [reply pom.xml:12]: yes lets upgrade!

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        <groupId>com.coveo</groupId>
+        <artifactId>fmt-maven-plugin</artifactId>
+        <version>2.10</version>
+        <executions>
+            <execution>
+                <goals>
+                    <goal>check</goal>
+                </goals>
+            </execution>
+        </executions>
```
Comment (gracetian6) [inline pom.xml:60]: Not sure what this is doing?

**Comment by laptou** [comment]
I changed my primary email address to `ibiyemi@google.com` for now, so you can just commit the patches.

File: pom.xml (hunk 5 above)
Comment (seunomonije) [reply pom.xml:60]: It's just a part of the plugin

**Comment by seunomonije** [comment]
It keeps screwing everything up when I commit inline. I'll just close this one and make a fresh one..it's getting too cluttered

**Pull request closed without merging**